mobile/test/common/integration/xds_integration_test.cc

```diff
@@ -27,20 +72,17 @@ XdsIntegrationTest::XdsIntegrationTest() : BaseClientIntegrationTest(ipVersion()
     config_helper_.addRuntimeOverride("envoy.reloadable_features.unified_mux", "true");
   }
 
-  // Set up the basic bootstrap config for xDS.
+  // Callback to write the builder config to the bootstrap.
+  // This ensures the fake upstreams match what's been specified in the engine
   config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
-    // The default stats config has overenthusiastic filters.
-    bootstrap.clear_stats_config();
-
-    // Add two clusters by default:
-    //  - base_h2: An HTTP2 cluster with one fake upstream endpoint, for accepting requests from EM.
-    //  - xds_cluster.lyft.com: An xDS management server cluster, with one fake upstream endpoint.
-    bootstrap.mutable_static_resources()->clear_clusters();
-    bootstrap.mutable_static_resources()->add_clusters()->MergeFrom(
-        createSingleEndpointClusterConfig("base_h2"));
-    bootstrap.mutable_static_resources()->add_clusters()->MergeFrom(
-        createSingleEndpointClusterConfig(std::string(XDS_CLUSTER)));
+    auto config_str = builder_.generateConfigStr();
+    TestUtility::loadFromYaml(absl::StrCat(config_header, config_str), bootstrap);
   });
+  // Set up the basic bootstrap config for xDS.
+  // We use a fake port at first, and assign a real one during initialization
+  builder_.setCustomClusters(
+      xdsClustersConfig(Network::Test::getLoopbackAddressString(ipVersion()), 0, 0));
+  builder_.disableStatsConfig(true);
```

Comment:
Reposting an unresolved comment from Alyssa in the other repo:

"do you know what stat counters we need for xDS? maybe we should we just add them into the default matchers now we're adding builder xDS support?"

Might need tips on how to do this